reactor-core/src/main/java/reactor/core/publisher/ParallelFlux.java

```diff
@@ -1027,7 +1039,9 @@ public final Disposable subscribe(
 					new LambdaSubscriber<>(onNext, onError, onComplete, onSubscribe);
 		}
 
-		onLastAssembly(this).subscribe(subscribers);
+		@SuppressWarnings("unchecked")
+		MultiSubscriber<T> multiSubscriber = new MultiSubscriber<>((CoreSubscriber<T>[]) subscribers);
+		Hooks.onLastOperator(this).subscribe(multiSubscriber);
```

Comment:
can't we still just do `Hooks.onLastOperator(this).subscribe(subscribers)`?

Reply:
We can't, since `onLastOperator` returns `CorePublisher`, not `ParallelFlux` (that's the whole idea of this change)

Reply:
so the returned `MultiSubscriber` could be passed to a `Publisher` that doesn't understand this type at all and could call one of the methods that throw an UOE?

how about checking if transformed `CorePublisher` is `instanceof ParallelFlux` (which most of the time, ie when using liftFunction, it should) and passing the subscribers. And otherwise subscribe a single `LambdaSubscriber` anyway.